rfcs/react-components/convergence/slot-render-as.md

```diff
@@ -0,0 +1,155 @@
+# RFC: Use `as` instead of `children` for slot render function
+
+Contributors: behowell
+
+## Summary
+
+This proposes modifying the slot render function API to pass a render function via the slot's `as` prop instead of the `children` prop. This has several benefits, the primary of which is that it allows the user to write a render function for a slot without overriding the default children of a slot.
+
+This is a complementary proposal to bsunderhus's [RFC #27164: slot children render function support](https://github.com/microsoft/fluentui/pull/27164). It provides an alternative way to avoid having a render function override a slot's default `children` prop. This RFC has no dependencies on it, and could be adopted separately if needed.
+
+## Background
+
+Slots accept render functions that allow them to override the component and/or props used when rendering the slot. For more background, see https://react.fluentui.dev/?path=/docs/concepts-developer-customizing-components-with-slots--page#replacing-the-entire-slot
+
+The render function is passed in as the slot's `children` prop.
+
+## Problem statement
+
+A few problems with render functions as they are today:
+
+1. Using `children` as the render function will override any default children provided by the component's internals.
+2. It is confusing that passing a render function as `children` would override the _whole slot_, and not just its children.
+3. The root slot's render function conflicts the use of children as a function, which certain components use for custom rendering of their children.
+   - For example, Tooltip and Field both accept a function as a child of their root, in order to pass props to the children. However, this breaks the ability to use a slot render function for the root slot of those controls.
```

Comment:
Can you please add an example?

Reply:
The example is in Pros 2. I can copy that example up here to explain why it's broken before this change.